petastorm/arrow_reader_worker.py

```diff
@@ -173,6 +173,10 @@ def process(self, piece_index, worker_predicate, shuffle_row_drop_partition):
     def _check_shape_and_ravel(x, field):
         if not isinstance(x, np.ndarray):
             raise ValueError('field {name} must be numpy array type.'.format(name=field.name))
+        if None in field.shape:
```

Comment:
Should we also check dim value -1 in shape ?

Reply:
Not sure. `None` is a generally, valid element in UnischemaField's shape field, but -1 is not. If some validation on invalid range on dimensions should be done, I would think that it should be done sometime during the construction of UnischemaField?